pandas/_testing/asserters.py

```diff
@@ -1340,7 +1340,9 @@ def assert_indexing_slices_equivalent(ser: Series, l_slc: slice, i_slc: slice) -
         assert_series_equal(ser[l_slc], expected)
 
 
-def assert_metadata_equivalent(left, right) -> None:
+def assert_metadata_equivalent(
+    left: DataFrame | Series, right: DataFrame | Series = None
```

Comment:
almost correct, as `right` can be `None` as well, so you need to include that in the type annotations (`| Series | None`)